instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/kafka/utils.py

```diff
@@ -0,0 +1,202 @@
+        return kwargs.get(key, default_value)
+
+    @staticmethod
+    def extract_send_topic(args: Any, kwargs: Any):
+        """extract topic from `send` method arguments in KafkaProducer class"""
+        return KafkaPropertiesExtractor._extract_argument(
+            "topic", 0, "unknown", args, kwargs
+        )
+
+    @staticmethod
+    def extract_send_value(args: Any, kwargs: Any):
+        """extract value from `send` method arguments in KafkaProducer class"""
+        return KafkaPropertiesExtractor._extract_argument(
+            "value", 1, None, args, kwargs
+        )
+
+    @staticmethod
+    def extract_send_key(args: Any, kwargs: Any):
+        """extract key from `send` method arguments in KafkaProducer class"""
+        return KafkaPropertiesExtractor._extract_argument("key", 2, None, args, kwargs)
+
+    @staticmethod
+    def extract_send_headers(args: Any, kwargs: Any):
+        """extract headers from `send` method arguments in KafkaProducer class"""
+        return KafkaPropertiesExtractor._extract_argument(
+            "headers", 3, None, args, kwargs
+        )
+
+    @staticmethod
+    def extract_send_partition(args: Any, kwargs: Any):
+        return KafkaPropertiesExtractor._extract_argument(
+            "partition", 4, None, args, kwargs
+        )
+
+    @staticmethod
+    def extract_client_id(client: AIOKafkaClient):
+        return client._client_id
+
+
+ProduceHookT = Optional[Callable[[Span, List, Dict], None]]
+ConsumeHookT = Optional[Callable[[Span, ABCRecord, List, Dict], None]]
+
+
+class KafkaContextGetter(textmap.Getter[textmap.CarrierT]):
+    def get(self, carrier: textmap.CarrierT, key: str) -> Optional[List[str]]:
+        if carrier is None:
+            return None
+
+        for item_key, value in carrier:
+            if item_key == key:
+                if value is not None:
+                    return [value.decode()]
+        return None
+
+    def keys(self, carrier: textmap.CarrierT) -> List[str]:
+        if carrier is None:
+            return []
+        return [key for (key, value) in carrier]
+
+
+class KafkaContextSetter(textmap.Setter[textmap.CarrierT]):
+    def set(self, carrier: textmap.CarrierT, key: str, value: str) -> None:
+        if carrier is None or key is None:
+            return
+
+        if value:
+            value = value.encode()
+        carrier.append((key, value))
+
+
+_kafka_getter = KafkaContextGetter()
+_kafka_setter = KafkaContextSetter()
+
+
+def _enrich_span(
+    span: Span,
+    bootstrap_servers: List[str],
+    topic: str,
+    partition: Optional[int],
+    client_id: str,
+):
+    span.set_attributes(
+        {
+            SpanAttributes.MESSAGING_SYSTEM: "kafka",
+            SpanAttributes.MESSAGING_DESTINATION: topic,
+            SpanAttributes.MESSAGING_URL: json.dumps(bootstrap_servers),
+            SpanAttributes.MESSAGING_DESTINATION_KIND: "topic",
+            SpanAttributes.MESSAGING_KAFKA_CLIENT_ID: client_id,
```

Comment:
This should be `SpanAttributes.MESSAGING_CLIENT_ID` as `MESSAGING_KAFKA_CLIENT_ID` has been deprecated